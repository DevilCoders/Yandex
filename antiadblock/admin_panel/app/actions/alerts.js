import servicesApi from 'app/api/services';
import {setErrors, clearErrors} from 'app/actions/errors';
import {getServices, getErrors} from 'app/reducers/index';
import {getById} from 'app/reducers/errors';
import {timeIsOver} from 'app/lib/errors-logic';

export const START_ALERTS_LOADING = 'START_ALERTS_LOADING';
export const END_ALERTS_LOADING = 'END_ALERTS_LOADING';

export function fetchAlerts() {
    return (dispatch, getState) => {
        dispatch(startAlertsLoading());

        return servicesApi.fetchAlerts().then(alerts => {
            const errors = getById(getErrors(getState()), 'dashboard');
            if (errors && errors.length) {
                dispatch(clearErrors('dashboard'));
            }

            return dispatch(endAlertsLoading(alerts.items));
        }, error => {
            const lastUpdateTime = getServices(getState()).lastUpdateTimeAlerts;

            if (timeIsOver(lastUpdateTime)) {
                return dispatch(setErrors('dashboard', [error.message]));
            }
        });
    };
}

export function startAlertsLoading() {
    return {
        type: START_ALERTS_LOADING
    };
}

export function endAlertsLoading(items) {
    return {
        type: END_ALERTS_LOADING,
        items
    };
}
