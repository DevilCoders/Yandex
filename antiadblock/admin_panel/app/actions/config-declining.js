export const OPEN_CONFIG_DECLINING = 'OPEN_CONFIG_DECLINING';
export const CLOSE_CONFIG_DECLINING = 'CLOSE_CONFIG_DECLINING';

export function openConfigDeclining(serviceId, labelId, id) {
    return {
        type: OPEN_CONFIG_DECLINING,
        serviceId,
        labelId,
        id
    };
}

export function closeConfigDeclining(id) {
    return {
        type: CLOSE_CONFIG_DECLINING,
        id
    };
}

