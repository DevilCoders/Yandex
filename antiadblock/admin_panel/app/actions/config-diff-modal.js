export const OPEN_CONFIG_DIFF_MODAL = 'OPEN_CONFIG_DIFF_MODAL';
export const CLOSE_CONFIG_DIFF_MODAL = 'CLOSE_CONFIG_DIFF_MODAL';

export function openConfigDiffModal(serviceId, id, secondId) {
    return {
        type: OPEN_CONFIG_DIFF_MODAL,
        serviceId,
        id,
        secondId
    };
}

export function closeConfigDiffModal() {
    return {
        type: CLOSE_CONFIG_DIFF_MODAL
    };
}
