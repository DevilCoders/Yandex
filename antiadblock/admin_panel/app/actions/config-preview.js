export const OPEN_CONFIG_PREVIEW = 'OPEN_CONFIG_PREVIEW';
export const CLOSE_CONFIG_PREVIEW = 'CLOSE_CONFIG_PREVIEW';

export function openConfigPreview(serviceId, id, configData) {
    return {
        type: OPEN_CONFIG_PREVIEW,
        serviceId,
        id,
        configData
    };
}

export function closeConfigPreview(id) {
    return {
        type: CLOSE_CONFIG_PREVIEW,
        id
    };
}
