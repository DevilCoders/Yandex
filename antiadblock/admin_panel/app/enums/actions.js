export const ACTIONS = {
    // Временно скрыт мониторинг на сервисах, т.к. он пока не работает
    // GRAPHS: 'graphs',
    CONFIGS: 'configs',
    AUDIT: 'audit',
    INFO: 'info',
    UTILS: 'utils',
    // ANTIADB-2636
    // HEALTH: 'health',
    RULES: 'rules',
    SBS_RESULT_CHECKS: 'sbs-result-checks'
};

export const SECRET_ACTIONS = {
    SCREENSHOTS: 'screenshots-checks',
    PROFILES: 'sbs-profiles',
    LOGS: 'logs'
};
