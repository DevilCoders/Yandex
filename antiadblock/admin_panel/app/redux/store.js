import {createStore, compose, applyMiddleware} from 'redux';
import reducer from 'app/reducers/index';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';

let composeEnhancers = compose;

export function configureStore(state = {}) {
    if (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION__) {
        composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;
    }
    if (process.env.NODE_ENV === 'local') {
        return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware, createLogger())));
    }
    return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware)));
}
