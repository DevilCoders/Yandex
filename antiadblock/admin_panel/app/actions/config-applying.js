import configApi from 'app/api/config';
import {setGlobalErrors} from 'app/actions/errors';

export const OPEN_CONFIG_APPLYING = 'OPEN_CONFIG_APPLYING';
export const CLOSE_CONFIG_APPLYING = 'CLOSE_CONFIG_APPLYING';
export const START_ACTIVE_CONFIG_LOADING = 'START_ACTIVE_CONFIG_LOADING';
export const END_ACTIVE_CONFIG_LOADING = 'END_ACTIVE_CONFIG_LOADING';

export function openConfigApplying(serviceId, id, configData, target) {
    return {
        type: OPEN_CONFIG_APPLYING,
        serviceId,
        id,
        configData,
        target
    };
}

export function closeConfigApplying(id) {
    return {
        type: CLOSE_CONFIG_APPLYING,
        id
    };
}

export function fetchCurrentConfig(labelId, target) {
    return dispatch => {
        dispatch(startActiveConfigLoading(labelId, target));

        return configApi.fetchCurrentConfig(labelId, target).then(config => {
            return dispatch(endActiveConfigLoading(labelId, target, config));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startActiveConfigLoading(labelId, target) {
    return {
        type: START_ACTIVE_CONFIG_LOADING,
        labelId,
        target
    };
}

export function endActiveConfigLoading(labelId, target, config) {
    return {
        type: END_ACTIVE_CONFIG_LOADING,
        labelId,
        target,
        config
    };
}
