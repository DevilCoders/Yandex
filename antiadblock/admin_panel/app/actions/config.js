import configApi from 'app/api/config';
import {setGlobalErrors} from 'app/actions/errors';

import {antiadbUrl} from 'app/lib/url';
import redirect from 'app/lib/redirect';
import {getStatus} from 'app/lib/config-editor';

export const START_FETCH_CONFIG_LOADING = 'START_FETCH_CONFIG_LOADING';
export const END_FETCH_CONFIG_LOADING = 'END_FETCH_CONFIG_LOADING';
export const START_CONFIG_SAVING = 'START_CONFIG_SAVING';
export const END_CONFIG_SAVING = 'END_CONFIG_SAVING';

function prepareValidation(errors = []) {
    return errors.reduce((validation, error) => {
        switch (error.path[0]) {
            case 'comment':
                validation.comment.push(error);
                break;
            case 'data':
                if (!validation.data[error.path[1]]) {
                    validation.data[error.path[1]] = [];
                }

                validation.data[error.path[1]].push(error);
                break;
            default:
                break;
        }

        return validation;
    }, {
        comment: [],
        data: {}
    });
}

export function fetchConfigById(configId, status, expId) {
    return dispatch => {
        dispatch(startConfigLoading(configId));

        return configApi.fetchConfig(configId, status, expId).then(data => {
            return dispatch(endConfigLoading(configId, data));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function saveConfig(serviceId, labelId, parentId, comment, data, dataSettings) {
    return dispatch => {
        dispatch(startConfigSaving());

        return configApi.saveConfig(labelId, parentId, comment, data, dataSettings).then(() => {
            redirect(antiadbUrl(`/service/${serviceId}/label/${labelId}/`));
        }, error => {
            if (error.status === 400) {
                const properties = error.response.properties;
                const validation = prepareValidation(properties);

                return dispatch(endConfigSaving(validation));
            }
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startConfigSaving() {
    return {
        type: START_CONFIG_SAVING
    };
}

export function endConfigSaving(validation) {
    return {
        type: END_CONFIG_SAVING,
        validation
    };
}

export function startConfigLoading(configId) {
    return {
        type: START_FETCH_CONFIG_LOADING,
        configId
    };
}

function getConfigStatus(data) {
    const statuses = (data.statuses && data.statuses[0] && data.statuses[0].status && data.statuses.map(item => item.status)) || [];

    return getStatus(statuses, data.exp_id);
}

export function endConfigLoading(configId, data) {
    return {
        type: END_FETCH_CONFIG_LOADING,
        configId,
        data: {
            ...data,
            dataCurrent: data.data,
            status: getConfigStatus(data)
        }
    };
}
