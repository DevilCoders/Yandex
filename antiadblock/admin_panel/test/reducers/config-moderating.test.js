import reducer from 'app/reducers/service/moderating-config';
import * as configModeratingActionsFixtures from '../fixtures/actions/config-moderating';
import {START_SERVICE_CONFIG_APPROVING, END_SERVICE_CONFIG_APPROVING, START_SERVICE_CONFIG_DECLINING, END_SERVICE_CONFIG_DECLINING} from '../fixtures/store';

describe('config-moderating reducer', () => {
    const startState = START_SERVICE_CONFIG_APPROVING;
    const endState = END_SERVICE_CONFIG_APPROVING;

    it('should return the initial state', () => {
        expect(reducer(undefined, {})).toMatchSnapshot();
    });

    it('should handle START_SERVICE_CONFIG_MODERATING', () => {
        expect(
            reducer(startState, configModeratingActionsFixtures.START_SERVICE_CONFIG_APPROVING)
        ).toMatchSnapshot();
    });

    it('should handle END_SERVICE_CONFIG_MODERATING', () => {
        expect(
            reducer(endState, configModeratingActionsFixtures.END_SERVICE_CONFIG_APPROVING)
        ).toMatchSnapshot();
    });

    const startState2 = START_SERVICE_CONFIG_DECLINING;
    const endState2 = END_SERVICE_CONFIG_DECLINING;

    it('should return the initial state', () => {
        expect(reducer(undefined, {})).toMatchSnapshot();
    });

    it('should handle START_SERVICE_CONFIG_MODERATING', () => {
        expect(
            reducer(startState2, configModeratingActionsFixtures.START_SERVICE_CONFIG_DECLINING)
        ).toMatchSnapshot();
    });

    it('should handle END_SERVICE_CONFIG_MODERATING', () => {
        expect(
            reducer(endState2, configModeratingActionsFixtures.END_SERVICE_CONFIG_DECLINING)
        ).toMatchSnapshot();
    });
});
