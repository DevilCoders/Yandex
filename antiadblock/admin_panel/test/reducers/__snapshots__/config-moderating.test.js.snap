// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config-moderating reducer should handle END_SERVICE_CONFIG_MODERATING 1`] = `
Object {
  "approved": null,
  "comment": null,
  "id": null,
  "progress": false,
}
`;

exports[`config-moderating reducer should handle END_SERVICE_CONFIG_MODERATING 2`] = `
Object {
  "approved": null,
  "comment": null,
  "id": null,
  "progress": false,
}
`;

exports[`config-moderating reducer should handle START_SERVICE_CONFIG_MODERATING 1`] = `
Object {
  "approved": "approved",
  "comment": "",
  "configId": 239,
  "id": 239,
  "progress": true,
  "serviceId": 123,
  "type": "START_SERVICE_CONFIG_MODERATING",
}
`;

exports[`config-moderating reducer should handle START_SERVICE_CONFIG_MODERATING 2`] = `
Object {
  "approved": "declined",
  "comment": "decline message",
  "configId": 239,
  "id": 239,
  "progress": true,
  "serviceId": 123,
  "type": "START_SERVICE_CONFIG_MODERATING",
}
`;

exports[`config-moderating reducer should return the initial state 1`] = `
Object {
  "approved": null,
  "comment": null,
  "id": null,
  "progress": false,
}
`;

exports[`config-moderating reducer should return the initial state 2`] = `
Object {
  "approved": null,
  "comment": null,
  "id": null,
  "progress": false,
}
`;
